dev/archery/archery/integration/util.py

```diff
@@ -0,0 +1,81 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import os
+import six
+import string
+import subprocess
+import uuid
+
+import numpy as np
+
+
+def guid():
+    return uuid.uuid4().hex
+
+
+RANDS_CHARS = np.array(list(string.ascii_letters + string.digits),
+                       dtype=(np.str_, 1))
+
+# SKIP categories
+SKIP_ARROW = 'arrow'
+SKIP_FLIGHT = 'flight'
+
+ARROW_ROOT_DEFAULT = os.environ.get(
```

Comment:
Yank `ARROW_ROOT_DEFAULT` and pass an instance of ArrowSources around, see also https://github.com/apache/arrow/blob/master/dev/archery/archery/utils/source.py#L112-L144. Note that other archery sub-commands provide an `--src` option for this specifically, e.g. allowing the caller to specify the sources path.

Usage can be customized like this
```
# ArrowSources would be interfered from archery path.
archery integration
# ArrowSources points to /tmp/my-local-arrow-copy
archery integration --src=/tmp/my-local-arrow-copy
# ArrowSources points to /tmp/my-local-arrow-copy
ARROW_SRC=/tmp/my-local-arrow-copy archery integration
```

Thus the calling integration script can just copy the sources then export ARROW_SRC.

Reply:
You can rename ARROW_SRC with ARROW_HOME if you prefer, it is not used in scripts calling archery.

Reply:
I'll let this be done as follow up